baseplate.go

```diff
@@ -128,7 +128,12 @@ func Serve(ctx context.Context, server Server) error {
 			//
 			// This is a blocking call, so it is called in a separate goroutine.
 			go func() {
-				closeChannel <- server.Close()
+				bc := batchcloser.New(closers...)
+				be := errorsbp.Batch{}
```

Comment:
style nit: prefer `var be errorsbp.Batch`.

but `server` itself is also a closer, so it's actually easier to just add server into the batch closer, e.g.:

```
bc := batchcloser.New(closers...)
bc.Add(server)
closeChannel <- bc.Close()
```